applications/osb-portal/src/middleware/osbbackend.ts

```diff
@@ -77,23 +64,59 @@ const callAPIMiddlewareFn: Middleware = store => next => async (action: AnyActio
       UserService.register().then((user: any) => next({ ...action, payload: user }));
       break;
     case Workspaces.selectWorkspace.toString():
-      workspaceService.getWorkspace(action.payload).then(
-        (workspace: Workspace) => next({ ...action, payload: workspace }),
-        () => next(setError("Workspace not found or not accessible"))
-      );
-      break;
+
     case Workspaces.refreshWorkspace.toString():
-      workspaceService.getWorkspace(store.getState().workspaces.selectedWorkspace.id).then(
-        (workspace: Workspace) => next({ ...action, payload: workspace }),
-        () => next(setError("Workspace not found or not accessible"))
-      );
+
+      function refreshWorkspace(callback: (workspace: Workspace) => any) {
+        const selectedWorkspaceId = action.payload || store.getState().workspaces.selectedWorkspace.id;
+        refreshPending = true;
+        workspaceService.getWorkspace(selectedWorkspaceId).then(
+          (workspace: Workspace) => {
+            callback(workspace);
+            refreshPending = false;
+            if (!refreshPending && workspace.resources.find((resource: any) => resource.status === ResourceStatus.pending)) {
```

Comment:
strange construct --> !refreshPending is always true here (refreshPending is always false see line 76)

Reply:
imo the whole refreshpending is strange 
As I understand it you want to overcome that 2 refreshes take place at the same time
May be more cleaner is to:
1. stop the timer
2. try; exec commands; catch exceptions
3. start timer

Reply:
yes the purpose is to avoid more than one simultaneous auto refreshes, which was actually happening a lot, because we have not only the auto refresh but also other refreshes coming from the ui actions. And yes, ` refreshPending = false;` shoudn't be in that line.